.github/workflows/docker-base-images.yml

```diff
@@ -0,0 +1,74 @@
+name: "Docker base images"
+on:
+  # This GitHub Action will be invoked automatically from the main Azure DevOps build
+  # This job will only be launched when we are building the main branch (a snapshot) or when we are generating a release.
+  workflow_dispatch:
+    inputs:
+      azdo_build_id:
+        description: 'The specific AzDo build from which the release artifacts will be downloaded.'
+        required: true
+      is_snapshot:
+        description: 'True when we are building main branch, false when we are building a release'
+        required: true
+jobs:
+  build-and-publish-base-image:
+    runs-on: ubuntu-latest
+    env:
+      AZURE_DEVOPS_TOKEN: "${{ secrets.AZURE_DEVOPS_TOKEN }}"
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: '7.0.101'
+
+    - name: "Get current version"
+      id: versions
+      run: ./tracer/build.sh OutputCurrentVersionToGitHub
+
+    - name: "Download build assets from Azure Pipelines"
+      id: assets
+      run: ./tracer/build.sh DownloadAzurePipelineFromBuild 
+      env:
+        AzureDevopsBuildId: "${{ github.event.inputs.azdo_build_id }}"
+
+    - name: Copy tooling files to artifacts path
+      shell: bash
+      run: |
+        echo "OUTPUT ARTIFACT PATH: ${{steps.assets.outputs.artifacts_path}}"
+        cp ./tracer/build/_build/docker/reliability-env/* ${{steps.assets.outputs.artifacts_path}}
+
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v2
+
+    - name: Set up Docker Buildx
+      id: buildx
+      uses: docker/setup-buildx-action@v2
+
+    - name: Generate image tags
+      id: docker-base-image-tags
+      shell: bash
+      run: |
+        if [ "$is_snapshot" = "true" ]; then
+          echo "tag-names=ghcr.io/datadog/dd-trace-dotnet/dd-trace-dotnet:latest_snapshot" >> $GITHUB_OUTPUT
+        else
+          echo "tag-names=ghcr.io/datadog/dd-trace-dotnet/dd-trace-dotnet:latest" >> $GITHUB_OUTPUT
+        fi
+      env:
+        is_snapshot: "${{ github.event.inputs.is_snapshot}}"
+
+    - name: Login to Docker
+      run: docker login -u publisher -p ${{ secrets.GITHUB_TOKEN }} ghcr.io
+
+    - name: Docker Build linux-x64 image
```

Comment:
This only publishes an image for linux-x64, but we also support:
- linux-musl-x64 (linux-x64 but using musl)
- linux-arm64

What is the plan for different platforms? An image for each one?

Reply:
And while we're at it, what about Windows and macOS? (less important as not directly supported)